src/charm.py

```diff
@@ -198,6 +208,118 @@ def _nrpe_relation_broken(self, _):
         self._stored.have_nrpe = False
         self._set_status()
 
+    def _on_filebeat_relation_joined(self, event):
+        self._stored.have_filebeat = True
+
+        # Make sure the vector binary is present with a systemd service
+        if not Path("/usr/local/bin/vector").exists():
```

Comment:
Should this entire block (apart from systemd calls) go under `install` or charm lib?

Reply:
No, manipulating files on disk shouldn't be in the charm lib. As much as it's not actually handling relation data in this case.

We can't _really_ minimize resource utilization (it's not a subordinate), but like the exporter, we're treating this almost like socket-based activation and starting it as needed.